ksqldb-engine/src/main/java/io/confluent/ksql/services/KafkaConsumerGroupClientImpl.java

```diff
@@ -100,4 +105,23 @@ public ConsumerGroupSummary describeConsumerGroup(final String group) {
       throw new KafkaResponseGetFailedException("Failed to list Kafka consumer groups offsets", e);
     }
   }
+
+  @Override
+  public void deleteConsumerGroups(final Set<String> groups) {
+    final AtomicInteger retryCount = new AtomicInteger(0);
+    try {
+      // it takes heartbeat.interval.ms after a consumer is closed for the broker
+      // to recognize that there are no more consumers in the consumer group - for
+      // that reason, we retry after 3 seconds (the default heartbeat.interval.ms)
+      // in the case that we get a GroupNotEmptyException
+      ExecutorUtil.executeWithRetries(
+          () -> adminClient.get().deleteConsumerGroups(groups).all().get(),
+          e -> (e instanceof RetriableException)
+              || (e instanceof GroupNotEmptyException && retryCount.getAndIncrement() < 5),
+          () -> Duration.of(3, ChronoUnit.SECONDS)
```

Comment:
Why not use the actual value of `heartbeat.interval.ms`? Why hardcode it to 3 seconds?

Reply:
it's a consumer config so it can be different for each consumer (and unfortunately this isn't a constant in the kafka code, so I can't just use a constant from that class directly)